### PR TITLE
LwM2M: Ground fix object workaround and fixes

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Two changes in this PR:

### samples: lwm2m_client: Workaround ground fix problem
AVSystem server assumes incorrect object version for
Connection Monitoring object.

Breaks Ground Fix queries.
Force object version reporting by using different
Connection Monitor object version.

Reported as CDM-665

### net: lwm2m_client_utils: Forward ground fix data to Location obj

If LwM2M Location object is used and Ground Fix object receives
current location, forward that location to Location object as well.

### CODE REVIEW

**Review by juhaylinen — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -344,6 +344,10 @@ Cellular samples
 
   * Added the :ref:`CONFIG_UDP_DATA_UPLOAD_ITERATIONS <CONFIG_UDP_DATA_UPLOAD_ITERATIONS>` Kconfig option for configuring the number of data transmissions to the server.
 
+* :ref:`lwm2m_client` sample:
+
+  * Added workaround for Ground Fix Location assistance queries in AVSystem Coiote by using fixed Connectivity Monitor object version. This is enabled in :file:`overlay-assist-cell.conf` configuration overlay.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:349]:
```suggestion
  * Added a workaround for Ground Fix Location assistance queries in AVSystem Coiote by using fixed Connectivity Monitor object version.
  This is enabled in the :file:`overlay-assist-cell.conf` configuration overlay.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -684,6 +688,10 @@ Libraries for networking
 
   * Added support for using a password when connecting to a broker.
 
+* :ref:`lib_lwm2m_location_assistance` library:
+
+  * If device location is requested using Ground Fix location services, received coordinates will be copien in to the LwM2M Location object.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:693]:
```suggestion
  * Updated the received coordinated to be copied to the LwM2M Location object, when device location is requested using Ground Fix location services.
```
Or is it rather fixed than updated?

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -684,6 +689,10 @@ Libraries for networking
 
   * Added support for using a password when connecting to a broker.
 
+* :ref:`lib_lwm2m_location_assistance` library:
+
+  * Updated the Ground fix object to copy received coordinates to the LwM2M Location object.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:694]:
```suggestion
  * Updated the Ground Fix object to copy received coordinates to the LwM2M Location object.
```

**Review by peknis — APPROVED** [review]
Approved with minor nit.

**Comment by SeppoTakalo** [comment]
Added workaround for other overlays as well, as this seem to affect others as well. At least AGPS is affected.

**Review by trantanen — APPROVED** [review]

**Review by dilner — APPROVED** [review]

**Pull request merged**